### PR TITLE
Feat/search

### PR DESCRIPTION
**Pull request by andrewlew1s**
A quick search that looks through the names of the tools. Starts with an empty string as the default search, since all tools contain that they all show up and filtering works nice after.

Currently case sensitive, doesn't filter tags (perhaps we could have an option to filter by name vs tag? or just tag?) and ugly. 

Screen Shot 2020-03-15 at 11 57 36 PM
Screen Shot 2020-03-15 at 11 58 01 PM

### CODE REVIEW

**Comment by michaelfitzhavey** [comment]
This is part of issue #92 also.

**Comment by michaelfitzhavey** [comment]
Some early thoughts around this:
- lets use boostrap inputs like normal
- let's allow filtering by tag or description as well as name? (maybe?)

**Comment by andrewlew1s** [comment]
Ok so I looked into adding some radios as options for tag vs tool searching.

Thing is atm we are searching using a computed property, which doesn't seem to like logic like ifs as it needs to feel happy rendering itself first I guess. So i could change it to a method, but that might change how the search works, or I could maybe use a v-if to render two seperate computed tings based off the radio choice? 

What do you think @michaelfitzhavey ?

**Comment by michaelfitzhavey** [comment]
@andrewlew1s I see what you mean re: a radio but I think perhaps that is overcomplicating the UI for now.

Let's just have a single search term and do any fancy logic in the computed property `filteredTools`

File: website/src/Website.vue [hunk 1]
```diff
@@ -7,11 +7,17 @@
 				The Holistic Toolbox strives to provide a useful set of tools for developers, computer scientists and curious folk. With a growing array of diverse tools, your productivity can be increased in many fields. Built by a team of passionate and creative developers around the world who love helping others solve interesting problems.
 			</p>
 
+			<b-form-input type="text" v-model="search" placeholder="Search..." />
+			<b-form-radio-group id="radio-group" v-model="selected" name="radio-sub-component">
```
Comment (michaelfitzhavey) [inline website/src/Website.vue]: Let's remove this radio group for now (see below comment)

File: website/src/Website.vue [hunk 2]
```diff
@@ -31,6 +37,10 @@ export default {
 	},
 	data() {
 		return {
+			search: '',
+			selected: '',
```
Comment (michaelfitzhavey) [inline website/src/Website.vue]: let's remove `selected` along with the radio group

File: website/src/Website.vue [hunk 3]
```diff
@@ -31,6 +37,10 @@ export default {
...
+			searchTool: true,
```
Comment (michaelfitzhavey) [inline website/src/Website.vue]: as above remove with radio group

File: website/src/Website.vue [hunk 4]
```diff
@@ -31,6 +37,10 @@ export default {
...
+			searchTag: false,
```
Comment (michaelfitzhavey) [inline website/src/Website.vue]: as above remove with radio group

File: website/src/Website.vue [hunk 5]
```diff
@@ -64,7 +74,8 @@ export default {
 					name: 'Markdown Renderer',
 					description: 'A browser based markdown renderer.',
 					link: 'https://markdown-renderer.holistic-toolbox.com',
-					image: require('../assets/tools/markdown-renderer.png')
+					image: require('../assets/tools/markdown-renderer.png'),
+					tags: ['TEST']
```
Comment (michaelfitzhavey) [inline website/src/Website.vue]: remember to remove this before merging

File: website/src/Website.vue [hunk 6]
```diff
@@ -86,6 +97,12 @@ export default {
 				}
 			]
 		};
+	},
+	computed: {
+		filteredTools() {
+			return this.tools.filter(tool => tool.name.toLowerCase().match(this.search.toLowerCase()));
```
Comment (michaelfitzhavey) [inline website/src/Website.vue]:
```suggestion
			return this.tools.filter(tool => {
				const matchName = tool.name.toLowerCase().match(this.search.toLowerCase());
				const matchTags = tool.tags.includes(this.search.toUpperCase());
				return matchName || matchTags;
			});
```

Comment (andrewlew1s) [reply website/src/Website.vue]: so this is how you do that

Comment (michaelfitzhavey) [reply website/src/Website.vue]: I think so, YMMV - let me know how you get on with that

Comment (andrewlew1s) [reply website/src/Website.vue]: Went with having to map tool.tags as it is an array, however some of tools dont have tags so it gets an underfined error. I sorted this by wrapping `const matchTags` in an `if(tool.tags){}` but then everything broke. Decided enough time spent on error, will try again soon/ maybe we can figure it out real quick one time. 

`return this.tools.filter(tool => {
				const matchName = tool.name.toLowerCase().match(this.search.toLowerCase());
				const matchTags = tool.tags.map(s => s.toLowerCase().match(this.search.toLowerCase()));
				return matchName || matchTags;
			});`

Comment (michaelfitzhavey) [reply website/src/Website.vue]: `map` doesn't make sense here as `map` means "run a function for each item in the array", which isn't what we are trying to do.

This PR should clarify things a lil: https://github.com/holistic-web/toolbox/pull/121

Comment (andrewlew1s) [reply website/src/Website.vue]: I think I was misdirected by a stackoverflow rather than actually trying to do anything specific lol

**Pull request closed without merging**